src/Generating/FinishGen.h

```diff
@@ -70,6 +70,71 @@ class cFinishGenNetherClumpFoliage :
 
 
 
+class cFinishGenClumpTopBlock :
+	public cFinishGen
+{
+public:
+	// Contains the meta, type and weight for a clump block
+	struct BlockInfo
```

Comment:
Well, at first it was all FoliageXYZ instead of Block, but then I rebuild it in a way that it can be used for other things than flowers.